MergerService/Utils/JobUtils.cs

```diff
@@ -0,0 +1,63 @@
+using MergerLogic.Clients;
+using System.Diagnostics;
+using MergerLogic.Utils;
+using MergerService.Controllers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace MergerService.Utils
+{
+    // TODO: rename all utils to clients
+    public class JobUtils : IJobUtils
+    {
+        private readonly IHttpRequestUtils _httpClient;
+        private readonly IConfigurationManager _configuration;
+        private readonly ILogger _logger;
+        private readonly IHeartbeatClient _heartbeatClient;
+        private readonly ActivitySource _activitySource;
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+        private readonly string _jobManagerUrl;
+
+        public JobUtils(IConfigurationManager configuration, IHttpRequestUtils httpClient, ILogger<JobUtils> logger,
+            ActivitySource activitySource, IHeartbeatClient heartbeatClient)
+        {
+            this._httpClient = httpClient;
+            this._configuration = configuration;
+            this._logger = logger;
+            this._heartbeatClient = heartbeatClient;
+            this._activitySource = activitySource;
+
+            // Construct Json serializer settings
+            _jsonSerializerSettings = new JsonSerializerSettings();
+            _jsonSerializerSettings.Converters.Add(new StringEnumConverter());
+
+            _jobManagerUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
+            //TODO: add tracing
+        }
+
+        public MergeJob? GetJob(string jobId)
+        {
+            using (this._activitySource.StartActivity("dequeue task"))
+            {
+                string relativeUri = $"jobs/{jobId}";
+                string url = new Uri(new Uri(_jobManagerUrl), relativeUri).ToString();
+                string? jobData = this._httpClient.GetDataString(url);
+                if (jobData is null)
+                {
+                    this._logger.LogInformation($"Job id:{jobData}, not found");
+                    return null;
+                }
+
+                try
+                {
+                    return JsonConvert.DeserializeObject<MergeJob>(jobData, this._jsonSerializerSettings)!;
+                }
+                catch (Exception e)
+                {
+                    this._logger.LogWarning(e.Message);
```

Comment:
i think it should be an error level message

Reply:
Done